src/lib/containers/oauth/OAuthConfirmationModal.tsx

```diff
@@ -0,0 +1,75 @@
+import React from 'react'
+import { Button, Modal } from 'react-bootstrap'
+import {
+  useDeleteOAuthClient,
+  useMutateOAuthClient,
+} from '../../utils/hooks/SynapseAPI'
+import { OAuthClient } from '../../utils/synapseTypes/OAuthClient'
+import Typography from '../../utils/typography/Typography'
+import { displayToast } from '../ToastMessage'
+import { AlertIcon } from '../../assets/icons/Alert'
+
+export type ConfirmModalProps = {
```

Comment:
Prefer to use WarningModal instead of this custom one-off component, see comment on CreateOAuthClient.tsx

Reply:
Kind of unusual to have all this much logic in the modal when it could be captured in handlers passed to the modal